Closes 1403: Added style for Natural Image behavior

cpoff opened the pull request:
This issue in #1403 was when you embed an image, set the size to Natural, and align center ... the block aligned left. So I added a wrapper class to override that display setting and make it `flex` instead of `block`. Changes should apply only to the Natural Image display. Did a new class because I didn't want to bother existing blocks using that behavior.

It required changes to two files:

1. Structure > Media Type > Image (Edit) > Manage display (Filename: modules/custom/az_media/config/install/core.entity_view_display.media.az_image.az_natural_size.yml) to add new container/wrapper setting the class to align-image-natural.
2. Added those class settings to style.css:

```
.align-natural-image {
  display:flex;
}
```

trackleft commented:
You need to add this config 
`core.entity_view_display.media.az_image.az_natural_size`

```
160
161In UnmetDependenciesException.php line 100:
162                                                                               
163  [Drupal\Core\Config\UnmetDependenciesException]                              
164  Configuration objects provided by <em class="placeholder">az_media</em> hav  
165  e unmet dependencies: <em class="placeholder">core.entity_view_display.medi  
166  a.az_image.az_natural_size (image.style.natural)</em>                        
```

cpoff commented:
I added image.style.az_natural.yml, but it has some SecKit formatting that I can't figure out. Here's an example of the Small image config style that I cloned:

```
langcode: en
status: true
name: az_small
label: 'Small (360px)'
effects:
  a5aee9bc-6dd3-4b48-b04e-027cc8017630:
    uuid: a5aee9bc-6dd3-4b48-b04e-027cc8017630
    id: image_scale
    weight: 1
    data:
      width: 360
      height: null
      upscale: true
```

trackleft commented:
Alternatively we could add another class to set `width: fit-content `if we didn't want to alter the align-center class

akslay commented:
> Alternatively we could add another class to set `width: fit-content `if we didn't want to alter the align-center class

I like the idea of adding a utility class of something like `.w-fit-content` (to match other width utility classes like `.w-100`). And then have that class set the `width: fit-content` style.

As a side note, the width utility classes are coming from Bootstrap. So it might be a good idea to put it in there rather than in Quickstart, but that would end up delaying this PR. So maybe we can put it in QA CSS for now and have a follow-up to get it added to AZ Bootstrap?

trackleft commented:
> As a side note, the width utility classes are coming from Bootstrap. So it might be a good idea to put it in there rather than in Quickstart, but that would end up delaying this PR. So maybe we can put it in QS CSS for now and have a follow-up to get it added to AZ Bootstrap?

I think we can do what you described and make a follow-up ticket in arizona-bootstrap.

cpoff commented:
Just for the record, here's my existing process:

Add new wrapper in `Media Types > Image > Display Type` to add "natural-image-align" class around the entire figure:

```
.natural-image-align {
	display:grid;
}
```

.... added to the custom theme style.css.

This was done with a custom class as to not clobber the `align-center` class, which specifies "display:block" and is used elsewhere around the site.

Plus I added the new image style, which should draw the default `img-fluid` class applied to all images with a defined image style.

trackleft commented:
FYI another alternative would be to use the style name prefix that gets added to the img element.

```
.az-natural-size.img-fluid {
  width: fit-content'
}
```

trackleft commented:
FYI https://caniuse.com/mdn-css_properties_width_fit-content

trackleft commented:
FYI there is no need to add a new class, `d-flex` would change the display

https://digital.arizona.edu/arizona-bootstrap/docs/2.0/utilities/display/

cpoff commented:
Indeed it does. Thanks Chris!

Would it be acceptable to re-add a line back in the Media Type > Natural Display? And rather than creating a new class/adding definition to style.css just include that `d-flex` class in a standalone div?

cpoff commented:
![Screen Shot 2022-04-14 at 4 08 39 PM](https://user-images.githubusercontent.com/730076/163491624-bf64e135-06e4-47ea-af58-0b011fde370c.png)

cpoff commented:
`d-flex` utility class installed. Ready for checkmarks.

joeparsons commented on themes/custom/az_barrio/css/style.css:
```diff
@@ -6,11 +6,20 @@
  * of these @include files will be combined into a single file.
  */
 
+/* Class for Natural Image block alignment issue */
+.align-natural-image {
+  display:flex;
+}
+
 /* For structuring header utilities*/
 header#header_arizona {
   min-height: 10px;
 }
 
+.align-natural-image {
+  display:flex;
+}
+
```
Should these styles get removed now that you're using the `d-flex` class?

```suggestion
/* For structuring header utilities*/
header#header_arizona {
  min-height: 10px;
}
```

smayden reviewed (approved):
tested as content admin. natural style can now be centered. Other image styles are still working.

cpoff commented:
style.css updated to remove old styles. Ready to merge.

danahertzberg commented on themes/custom/az_barrio/css/style.css:
```diff
@@ -6,6 +6,7 @@
  * of these @include files will be combined into a single file.
  */
 
+
```
Do we care about the extra added space here? 😬

danahertzberg replied:
```suggestion
```

joeparsons replied:
would be better to remove it, if possible.
```suggestion
```

danahertzberg commented on themes/custom/az_barrio/css/style.css:
```diff
@@ -6,6 +6,7 @@
  * of these @include files will be combined into a single file.
  */
 
+
```
```suggestion
```
